domenetjenester/iay/src/main/java/no/nav/foreldrepenger/abakus/registerdata/inntekt/sigrun/klient/SigrunConsumerImpl.java

```diff
@@ -94,6 +100,22 @@ public SigrunSummertSkattegrunnlagResponse summertSkattegrunnlag(Long aktørId,
         }
     }
 
+    private IntervallEntitet justerOpplysningsperiodeNårSisteÅrIkkeErFerdiglignet(Long aktørId, IntervallEntitet opplysningsperiode) {
+        // justerer slik at vi henter ett å eldre data når siste år som etterspørs ikke er ferdiglignet enda
+        int fomÅr = opplysningsperiode.getFomDato().getYear();
+        int tomÅr = opplysningsperiode.getTomDato().getYear();
+        Year iFjor = Year.now().minusYears(1);
+        if (opplysningsperiode.getTomDato().getYear() == iFjor.getValue() && !iFjorErFerdiglignetBeregnet(aktørId, iFjor)) {
+            tomÅr--; //justerer slutten, har ikke ferdiglignet data for året
+            if (tomÅr - fomÅr < 2) { //dataminimering, ikke behov for data ut over 3 hele år før første stp
+                fomÅr--;
+            }
+        }
+        LocalDate fom = LocalDate.of(fomÅr, 1, 1);
+        LocalDate tom = LocalDate.of(tomÅr, 12, 31);
+        return IntervallEntitet.fraOgMedTilOgMed(fom, tom);
```

Comment:
Bør du ikke beholde tomÅr as-is fra request og heller redusere fomÅr med ett år ?
Det går på at ferdigligninget pågår fom main til desember - 15% ila juni, 40% ila juli, 75% ila september, 85% ila oktober, osv. Du vet ikke om fjoråret er ferdiglignet eller ei når du står på en gitt dato

Reply:
Det er absolutt unødvendig å flytte tomÅr. Her sjekker vi rett før om fjoråret er ferdiglignet, så det bør ikke være noe problem egentlig.